packages/next/plugins/with-nx.ts

```diff
@@ -197,6 +197,12 @@ function withNx(
             : joinPathFragments(outputDir, '.next');
       }
 
+      // If we are running a static serve of the Next.js app, we need to change the output to 'export' and the distDir to 'out'.
```

Comment:
would it lead to unexpected behavior if file server executor got killed in between?

Reply:
No it doesn't change the source file and if the file serve executor is killed this is child process and would be killed as well.